view.go

```diff
@@ -206,45 +205,45 @@ func (v *view) recalculateCaches() {
 
 // CreateFragmentIfNotExists returns a fragment in the view by shard.
 func (v *view) CreateFragmentIfNotExists(shard uint64) (*fragment, error) {
-	frag, msg, err := v.createFragmentIfNotExists(shard)
-
-	// if msg is not nil, then a new shard was created
-	if err == nil && msg != nil {
-		// Broadcast a message that a new max shard was just created.
-		if err = v.broadcaster.SendSync(msg); err != nil {
-			frag.close()
-			return nil, errors.Wrap(err, "sending createshard message")
-		}
-		v.mu.Lock()
-		v.fragments[shard] = frag
-		v.mu.Unlock()
-	}
-
-	return frag, err
-}
-
-func (v *view) createFragmentIfNotExists(shard uint64) (*fragment, *CreateShardMessage, error) {
 	v.mu.Lock()
 	defer v.mu.Unlock()
 	// Find fragment in cache first.
 	if frag := v.fragments[shard]; frag != nil {
-		return frag, nil, nil
+		return frag, nil
 	}
 
 	// Initialize and open fragment.
 	frag := v.newFragment(v.fragmentPath(shard), shard)
 	if err := frag.Open(); err != nil {
-		return nil, nil, errors.Wrap(err, "opening fragment")
+		return nil, errors.Wrap(err, "opening fragment")
 	}
 	frag.RowAttrStore = v.rowAttrStore
 
-	msg := &CreateShardMessage{
-		Index: v.index,
-		Field: v.field,
-		Shard: shard,
+	v.fragments[shard] = frag
+	broadcastChan := make(chan struct{})
+
+	go func() {
+		msg := &CreateShardMessage{
+			Index: v.index,
+			Field: v.field,
+			Shard: shard,
+		}
+		// Broadcast a message that a new max shard was just created.
+		err := v.broadcaster.SendSync(msg)
+		if err != nil {
+			v.logger.Printf("broadcasting create shard: %v", err)
+		}
+		close(broadcastChan)
+	}()
+
+	// We want to wait until the broadcast is complete, but what if it
+	// takes a really long time? So we time out.
+	select {
+	case <-broadcastChan:
+	case <-time.After(50 * time.Millisecond):
```

Comment:
consider logging a warning here? sendSync is taking a long time, other nodes may not know about this shard, results may not be consistent until it finishes... etc

Reply:
I'd be a bit concerned about it being spammy; I don't know how often we'd expect that to happen, or how often new shards get created. I'd say maybe... Make a version that does that and do some testing? But I'm not sure what to test it on, for workloads.

Reply:
make it a logger.Debug?